### DISCUSSION TITLE
Elsa Core Expecting ElsaDBContextOptions

### DISCUSSION THREAD

**Discussion by vinujosey** [elsa-workflows/elsa-core#5660] [Q&A] [answered]
Hi when using the below extensions for entityframework, Elsa core is expecting the DBcontext to be ElsaDBContext, Is there a way I can use my db context instead. 

builder.Services.AddElsa(elsa =>
{
    elsa.UseWorkflowManagement(management => management.UseEntityFrameworkCore(x => x.UseSqlServer(dbconnection)));

    elsa.UseWorkflowRuntime(runtime => runtime.UseEntityFrameworkCore(x => x.UseSqlServer(dbconnection)));

});

**Comment by glime-ai** [comment, marked as answer]
To use your own DbContext with Elsa Workflows and Entity Framework Core, you need to ensure that your DbContext derives from ElsaContext or includes the necessary DbSet<T> properties that Elsa requires. You can then configure Elsa to use your custom DbContext by specifying it in the `AddEntityFrameworkStores` method.

Here's an example of how you might configure Elsa to use your custom DbContext:

```csharp
public class MyCustomDbContext : ElsaContext
{
    public MyCustomDbContext(DbContextOptions options) : base(options)
    {
    }

    // Your custom DbSets...
}

// ...

builder.Services.AddElsa(elsa =>
{
    // Use your custom DbContext for the management layer.
    elsa.UseWorkflowManagement(management => management.UseEntityFrameworkCore<MyCustomDbContext>());

    // Use your custom DbContext for the runtime layer.
    elsa.UseWorkflowRuntime(runtime => runtime.UseEntityFrameworkCore<MyCustomDbContext>());
});
```

If you want to keep your DbContext separate and not derive from ElsaContext, you can add the necessary Elsa DbSet<T> properties to your DbContext and then configure Elsa to use it:

```csharp
public class MyCustomDbContext : DbContext
{
    public MyCustomDbContext(DbContextOptions options) : base(options)
    {
    }

    // Your custom DbSets...
    public DbSet<WorkflowInstance> WorkflowInstances { get; set; }
    public DbSet<WorkflowDefinition> WorkflowDefinitions { get; set; }
    // Other Elsa-related DbSets...
}

// ...

builder.Services.AddElsa(elsa =>
{
    // Use your custom DbContext for the management layer.
    elsa.UseWorkflowManagement(management => management.UseEntityFrameworkCore<MyCustomDbContext>());

    // Use your custom DbContext for the runtime layer.
    elsa.UseWorkflowRuntime(runtime => runtime.UseEntityFrameworkCore<MyCustomDbContext>());
});
```

Make sure to replace `MyCustomDbContext` with the actual name of your DbContext and configure it with the necessary connection string or options.

Remember to apply any required migrations to create the necessary tables for Elsa in your database. You can do this using the `Update-Database` command and specifying your DbContext.

If you encounter issues with multiple DbContexts or need to specify a particular one for migrations, you can use the `-Context` parameter with the `Update-Database` command in the Package Manager Console or `--context` in the .NET CLI.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
